### ISSUE TITLE
CoreDisplay broken after unsuccessful patch

### ISSUE THREAD

**Issue by andsimakov** [Floris497/mac-pixel-clock-patch-V2#224]
I tried to apply CoreDisplay patch to my Mac mini late 2012/macOS 10.12.6 but something had gone wrong. After reboot I always have frozen progress bar and frozen round spinner below Apple logo. Mac mini is still accessible via AFP but not via VNC.

I checked MD5 of its CoreDisplay and same file from my MacBookPro 13'' 2015 with same 10.12.6. They have different checksum.

The issue is that I'm not able to substitute damaged CoreDisplay at Mac mini with healthy one from MacBook Pro. What I did (SIP has been disabled of course):
— Boot in Recovery, went to Terminal
— Mounted / partition as writable (mount -uw /)
— Renamed damaged CoreDisplay to *.bak just in case
— Copied healthy CoreDisplay there

No luck.
— When I boot Mac mini to normal mode, I have frozen screen but CoreDisplay and CoreDisplay.bak via AFP on MacBook
— When I boot to Recovery on mini, I just can see CoreDisplay file and can't see CoreDisplay.bak as well as delete it?

What a ghost files/strange behavior? What secret I'm loosing here during work with system partition?

Thanks.

**Comment by Floris497** [comment]
No time to answer long now, just reinstall macOS takes about 20-40 min

**Comment by andsimakov** [comment]
Hm. Is this really special case with system partition? Just write key phrase to google. Reinstall isn't acceptable. Have a lot at system preinstalled. Or reinstalling doesn't touch anything 3rd party?

**Comment by Floris497** [comment]
Hm, actually idk, but reinstall should not touch "regular" 3rd party stuff.. maybe some scetchy stuff.. okay let me read the problem

**Comment by Floris497** [comment]
Hm, what file exactly are you renaming? You might be renaming an symbolic link (quick though)

**Comment by andsimakov** [comment]
Well. I see the situation for the moment. CoreDisplay now is broken, but CoreDisplay.bak (when I can see it) isn't that I created or copied. It looks like it is a backup created by patcher itself though I can't unpatch it in Terminal under Recovery Mode because it does not see CoreDisplay.bak :(

**Comment by andsimakov** [comment]
I see the symbolic links a bit upper tree but nope, I operate with ../Versions/A/CoreDisplay itself.

**Comment by andsimakov** [comment]
Fixed for the moment. The problems were:
— Terminal in Recovery Mode is kinda "disabled person". It can't work well with File System. So I can't rename/delete files even with SIP disabled and FS mounted as RW.
— Broken driver doesn't allow to launch as physical display (built-in/external) as well as via VNC so I'm not able to run any code in normal mode on Mac remotely
— I didn't have Remote Login option enabled on My mini and wasn't able to login via SSH

What helped? I had special Python script which fires each time upon Mac's reboot and even before somebody log ins to the system. I just added there 'mv' command and it renamed backup CoreDisplay without a problem.
